syntax = "proto3";

message ContractProcessed {
  // Address related items
  string address = 1;
  string name = 2;
  int64 created_timestamp = 3;
  string status = 4;
  bool is_token = 5;
  int64 contract_updated_block = 6;
  string contract_type = 7;
  string token_standard = 8;
  string symbol = 9;

  // Transaction related items
  string transaction_hash = 15;
  bool is_creation = 16;  // Bool for if creation or verification
}
